Assets/_BForBoss/_Weapons/Scripts/EquipmentBehaviour.cs

```diff
@@ -0,0 +1,160 @@
+using PerigonGames;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    public class EquipmentBehaviour : MonoBehaviour
+    {
+        [SerializeField] private InputActionAsset _inputActions;
+        [SerializeField] private WeaponBehaviour[] _weapons = null;
+        private int _currentWeaponIndex = 0;
+
+        private InputAction _reloadInputAction = null;
+        private InputAction _fireInputAction = null;
+        private InputAction _swapWeaponInputAction = null;
+        private bool _isMouseScrollEnabled = true;
+        
+        public void Initialize()
+        {
+            EnableEquipmentPlayerInput();
+            SetupWeapons();
+            SetupControllerSwapWeaponInputBinding();
+        }
+
+        private void EnableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Enable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Enable();
+            }
+
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Enable();
+            }
+
+            _isMouseScrollEnabled = true;
+        }
+        
+        private void DisableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Disable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Disable();
+            }
+            
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Disable();
+            }
+
+            _isMouseScrollEnabled = false;
+        }
+
+        private void SetupWeapons()
+        {
+            foreach (var weapon in _weapons)
+            {
+                weapon.Initialize(_fireInputAction, _reloadInputAction);
+                weapon.enabled = false;
+            }
+
+            _weapons[_currentWeaponIndex].enabled = true;
+        }
+        
+        private void SetupPlayerEquipmentInput()
+        {
+            _reloadInputAction = _inputActions.FindAction("Reload");
+            _fireInputAction = _inputActions.FindAction("Fire");
+            _swapWeaponInputAction = _inputActions.FindAction("WeaponSwap");
+        }
+
+        private void SetupControllerSwapWeaponInputBinding()
+        {
+            _swapWeaponInputAction.started += OnControllerSwapWeaponAction;
```

Comment:
Does this ever need to unsubscribe?

Reply:
technically no.  But i'll set it up OnDestroy()